lib/pbench/server/auth/auth.py

```diff
@@ -102,11 +97,15 @@ def get_user_id() -> Optional[str]:
         Returns:
             User ID of the authenticated user, None otherwise.
         """
-        user_id = None
         user = Auth.token_auth.current_user()
-        if user:
-            user_id = str(user.id)
-        return user_id
+        return str(user.id) if user else None
+
+    @staticmethod
+    def _get_secret_key() -> str:
+        """Returns the configured secret key."""
+        if not Auth.secret_key:
+            Auth.secret_key = os.getenv("SECRET_KEY", "my_precious")
+        return Auth.secret_key
```

Comment:
Consider making this a class method instead of a static one:
```
    @classmethod
    def _get_secret_key(cls) -> str:
        """Returns the configured secret key."""
        if not cls.secret_key:
            cls.secret_key = os.getenv("SECRET_KEY", "my_precious")
        return cls.secret_key
```
It's not a big difference, but, since the method is intended to access (and potentially write) a class attribute, it might as well be a class method, and it saves the method from having to know the name of the class.

Reply:
Good suggestion, but it is also being removed as a class altogether in PR #3114 (which will be updated and posted following this PR merging).